backend/test_observer/data_access/models.py

```diff
@@ -210,7 +210,7 @@ class TestExecution(Base):
 
     __tablename__ = "test_execution"
 
-    ci_link: Mapped[str | None] = mapped_column(String(200), nullable=True)
+    ci_link: Mapped[str | None] = mapped_column(String(200), nullable=True, unique=True)
     c3_link: Mapped[str | None] = mapped_column(String(200), nullable=True)
```

Comment:
shouldn't C3 link be unique as well?

Reply:
Yeah that's true. But I added uniqueness to ci_link as more of a means to add an index. Since we need to query test executions by ci_link. Currently, we don't need to query or filter by c3_link so there is no benefit of an index being there. So it feels like it would just add some extra cost on the db without benefit (as indexes are not free). Unless we want to enforce uniqueness which I technically don't see too much of an issue violating here (if it happens somehow). wdyt?

Reply:
are there potential errors we can avoid by enforcing c3_link to be unique?

Reply:
Hmm, right now we get this link from Jenkins (but we can later get it from C3). So suppose there was an error somewhere and two test executions were sent the same c3_link. If we have uniqueness constraint enforced then only one of them will have it, but that might not necessarily be the right one. The other will be empty. On the other hand, if we don't have uniqueness enforced then both will have the same link, which is an error too. I guess in all cases it's incorrect. So in that sense perhaps enforcing uniqueness isn't all that valuable